File: New folder/Contest 1/Get_the_shadow.py
class Solution:
    def solve(self, N, a):
        result = []
        i = 0

        while i < N:
            if a[i] != a[a[i] - 1]:
                a[a[i] - 1], a[i] = a[i], a[a[i] - 1]
            else:
                i += 1

        for i in range(N):
            if a[i] != i + 1:
                result.append(a[i])
                result.append(i + 1)
                break

        return result

File: New folder/Contest 1/Get_the_shadow.cpp
class Solution {
public:
    vector<int> solve(int N, vector<int>& a) {
        vector<int> result;
        int i = 0;

        while (i < N) {
            if (a[i] != a[a[i] - 1]) {
                swap(a[a[i] - 1], a[i]);
            } else {
                i++;
            }
        }

        for (int i = 0; i < N; i++) {
            if (a[i] != i + 1) {
                result.push_back(a[i]);
                result.push_back(i + 1);
                break;
            }
        }

        return result;
    }
};
